client/src/components/screens/Patient/components/FilesTab/index.tsx

```diff
@@ -1,233 +1,66 @@
-import React, { useState } from 'react';
+import './components/tableColumns'
+
+import React from 'react';
 import intl from 'react-intl-universal';
-import { useDispatch,useSelector } from 'react-redux';
-import { DownOutlined } from '@ant-design/icons';
-import { getPatientFileURL } from 'actions/patient';
+import { useSelector } from 'react-redux';
 import {
-Button,   Card, Dropdown, Menu,
-Table, 
-} from 'antd';
+  Card, Spin,Table} from 'antd';
 import { formatBytes } from 'helpers/file';
 import { ParsedPatientData } from 'helpers/providers/types';
-import { PatientResponse } from 'helpers/search/types';
+import { PatientResponse, UseFilesResponse } from 'helpers/search/types';
 import { State } from 'reducers';
 
-import { getFilesData } from 'store/graphql/files/actions';
+import { useFilesData } from 'store/graphql/files/actions';
 
 import EmptyCard from './components/EmptyCard';
-import MetadataModal from './metadataModal';
+import { filesColumns } from './components/tableColumns';
 
 import './styles.scss';
 
-const FilesTab = () => {
-  let previousData: any | null = null;
-  const dispatch = useDispatch();
-  const dataSource: any[] = [];
-  const patient = useSelector((state: State) => state.patient.patient.parsed) as ParsedPatientData;
-  const [isOpen, setIsOpenModal] = useState<boolean>(false);
-  const [documentReference, setDocumentReference] = useState<string>('');
-  let filesResults = getFilesData({ patientId: patient.id });
-
-  if (filesResults.loading) {
-    if (!filesResults.results && previousData) {
-      filesResults = previousData;
-    }
-    else{
-      return null
-    }
-  }
 
-  if (filesResults.results) {
-    previousData = filesResults;
-  }
+const makeRowsFromFilesResults = (filesResults:UseFilesResponse) => filesResults.results.docs.map((element: PatientResponse) => {
+    const {size} = element.content[0].attachment;
+    const url = {
+      file: element.content[0].attachment.url,
+      index: element.content.length > 1 ? element.content[1].attachment.url : '',
```

Comment:
are you sure it is not  lement.content.length > 0 ? your condition will work when there are at least 2 elements. Maybe that's what you want but pls make sure. If it is the case do not use a magic number. Do something like: const MIN_NUM_OF_WHATEVER =  ...